pkg/cmd/pulumi/util.go

```diff
@@ -839,3 +839,26 @@ func checkDeploymentVersionError(err error, stackName string) error {
 	}
 	return errors.Wrap(err, "could not deserialize deployment")
 }
+
+func getRefreshOption(proj *workspace.Project, refresh string) bool {
+	// we want to check for an explicit --refresh or a --refresh=true or --refresh=false
+	// no is the default we assign to the refresh default to be able to understand
+	// the difference between when the user actually interacted with the cli argument
+	// and the default functionality today
+	if refresh != "" && refresh != "no" {
+		refreshDetails, boolErr := strconv.ParseBool(refresh)
+		if boolErr != nil {
+			// the user has passed a --refresh but with a random value that we don't support
+			return false
+		}
+		return refreshDetails
+	}
+
+	// the user has not specifically passed an argument on the cli to refresh but has set a Project option to refresh
+	if proj.Options != nil && proj.Options.Refresh != nil && *proj.Options.Refresh == "always" {
```

Comment:
If we make `Refresh` a `string` instead of `*string` we can simplify this:

```suggestion
	if proj.Options != nil && proj.Options.Refresh == "always" {
```

Reply:
👍